### ISSUE TITLE
Occasional BSOD SYSTEM_SERVICE_EXCEPTION cldflt.sys

### ISSUE THREAD

**Issue by bavarialogy** [nextcloud/desktop#3053]
* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Expected behaviour
Normal system operation during Nextcloud VFS sync

### Actual behaviour
Every once in a while, the system crashes with a BSOD when having Nextcloud Desktop running in the background.

### Client configuration
Client version: Nextcloud-3.2.0-rc2-x64

Operating system: Windows 10 x64

OS language: English US

Installation path of client: default 

### Server configuration
Nextcloud version: NC 21.0.0

### Logs
I can provide minidumps and the Desktop client logfile via a Dropshare.

**Comment by allexzander** [comment]
Known issue with CfAPI. No known reason that can be fixed at the moment.

See:
https://community.spiceworks.com/topic/2237262-onedrive-s-cldflt-sys-causing-win-10-bsod-system_service_exception
https://answers.microsoft.com/en-us/windows/forum/all/bsod-systemserviceexception-cldfltsys/0d2d923b-07e3-4b22-bd4d-3eedcc9644e7

**Comment by MaxiMii05** [comment]
This happened to me when trying to open a large (8 GB) .rar file that was set to "free up space". The file got corrupted and showed "0 KB" on the server. [A reddit post I made about it.](https://www.reddit.com/r/NextCloud/comments/n651yz/virtual_files_just_corrupted_an_8gb_rar_file_of/)

**Comment by allexzander** [comment]
> This happened to me when trying to open a large (8 GB) .rar file that was set to "free up space". The file got corrupted and showed "0 KB" on the server. [A reddit post I made about it.](https://www.reddit.com/r/NextCloud/comments/n651yz/virtual_files_just_corrupted_an_8gb_rar_file_of/)

This is quite unfortunate. Any chance for us to have a look at client logs? @MaxiMii05

**Comment by MaxiMii05** [comment]
> Any chance for us to have a look at client logs?

Logging seems to have been turned off until now. The folder only has files from today. The incident happened yesterday.
To be honest, I'm not willing to try to reproduce the scenario.

> And, the server logs, if possible.

I reduced [the log](https://gist.github.com/MaxiMii05/6d96573ebc4b136e9f9955be0f46ca18) to the relevant time frame. Today, I am having other problems with the client that I can't accurately describe quite yet. As you can see, there are no entries for yesterday.

**Comment by allexzander** [comment]
@MaxiMii05 @bavarialogy 

Do you remember if, when trying to get files on demand (double-clicking or choosing "Make always available locally"), at some point, you tried to cancel the download of any of those files, and then, if you tried to download those files again without restarting the OS/Nextcloud app?

I was just able to reproduce the BSOD while canceling the download of the 4.1 GB file, so, it would be nice to know if that was the same issue that you've encountered. We're then will be working on a fix.

**Comment by allexzander** [comment]
@bavarialogy 

Could you please upload your logs here? https://cloud.nextcloud.com/s/CwYBRNwqHiBFWd5 (the password is your Github username)

**Comment by MaxiMii05** [comment]
> Do you remember if, when trying to get files on demand (double-clicking or choosing "Make always available locally"), at some point, you tried to cancel the download of any of those files, and then, if you tried to download those files again without restarting the OS/Nextcloud app?

The first time the BSOD happened, I only double-clicked the file (wich would normally download, then open in WinRAR). I did not cancel the download myself.

But yes, this may be similar to what happened the second time I encountered the BSOD. But it's a bit more complicated. 
1. I copied another version of the 8GB file out of the files_versions folder to my PC via SMB and renamed it to the original file name.
2. I deleted the corrupt file in the Nextcloud sync folder on my PC and put the restored version back, with the same name.
3. Nextcloud synced, I think. The SMB share now showed the true capacity of the file.
4. Nextcloud synced again. Now for the weird part: the file was being downloaded into the sync folder and simultaneously being uploaded to the server. Somewhere inbetween all of this, the Client gave me notifications "MOVE of file failed". _It looked like the transfer of the file on the server from the uploads folder to the proper directory failed. Maybe this is a weird conflict issue and independent of the BSOD._ The operation got stuck at 100%. The file and the temporary hidden file now both had the same size, but task manager still showed extremely high writing speeds. The hard drive screamed. 
5. When I tried anything, the Nextcloud Client and Windows File Explorer would crash.
6. After a forced taskkill of the Client and a restart of File Explorer, the Client would try the same procedure (4.) again. 
This repeated several times as I was trying to figure out what the problem was. I canceled one of the syncs. After that, I only right-clicked on the file and the PC crashed.

For now, I have "solved" the issue by putting the file on the ignore list. It's already on the server so this theoretically shouldn't make a difference. If I removed it from the ignore list, the Client would again download the file and get stuck at 100%.

This is from memory and I was under quite a bit of stress when all this happened, so some parts are probably inaccurate. I still hope this helps with fixing the problem.
